<launch>

    <node pkg="nodelet" type="nodelet" name="camera_manager" args="manager" output="screen"/>

    <node pkg="nodelet" type="nodelet" name="cv_camera" args="load cv_camera/CvCameraNodelet camera_manager" output="screen">
        <param name="frame_id" value="front_camera_optical"/>
        <param name="camera_info_url" value="file://$(find camera_rescaler)/camera_info/ideal_camera.yaml"/>
        <param name="device_id" value="0"/>
        <param name="rescale_camera_info" value="true"/> <!-- automatically rescale camera calibration info -->

        <!-- camera resolution, NOTE: camera_info file should match it -->
        <param name="image_width" value="640"/>
        <param name="image_height" value="480"/>
    </node>

    <node pkg="nodelet" type="nodelet" name="cv_camera_scaled" args="load camera_rescaler/CameraRescalerNodelet camera_manager" output="screen">
        <remap from="image_raw" to="cv_camera/image_raw"/>
        <remap from="camera_info" to="cv_camera/camera_info"/>
        <param name="target_width" value="320"/>
        <param name="target_height" value="240"/>
    </node>    

</launch>